app/src/main/java/org/wikipedia/notifications/PollNotificationService.kt

```diff
@@ -0,0 +1,106 @@
+            val lastNotificationTime =
+                response.query?.notifications?.list?.maxOfOrNull { it.utcIso8601 }.orEmpty()
+            if (lastNotificationTime > Prefs.remoteNotificationsSeenTime) {
+                Prefs.remoteNotificationsSeenTime = lastNotificationTime
+                retrieveNotifications(context)
+            }
+        }
+    }
+
+    private suspend fun retrieveNotifications(context: Context) {
+        NotificationPollBroadcastReceiver.DBNAME_WIKI_SITE_MAP.clear()
+        NotificationPollBroadcastReceiver.DBNAME_WIKI_NAME_MAP.clear()
+        val response =
+            ServiceFactory.get(WikipediaApp.getInstance().wikiSite).unreadNotificationWikis()
+        val wikiMap = response.query!!.unreadNotificationWikis
+        val wikis = mutableListOf<String>()
+        wikis.addAll(wikiMap!!.keys)
+        for (dbName in wikiMap.keys) {
+            if (wikiMap[dbName]!!.source != null) {
+                NotificationPollBroadcastReceiver.DBNAME_WIKI_SITE_MAP[dbName] =
+                    WikiSite(wikiMap[dbName]!!.source!!.base)
+                NotificationPollBroadcastReceiver.DBNAME_WIKI_NAME_MAP[dbName] =
+                    wikiMap[dbName]!!.source!!.title
+            }
+        }
+        getFullNotifications(context, wikis)
+    }
+
+    private suspend fun getFullNotifications(context: Context, foreignWikis: List<String?>) {
+        val notificationRepository = NotificationRepository(AppDatabase.getAppDatabase().notificationDao())
+
+        val response = ServiceFactory.get(WikipediaApp.getInstance().wikiSite)
+            .getAllNotifications(if (foreignWikis.isEmpty()) "*" else foreignWikis.joinToString("|"), "!read", null)
+        response.query?.notifications?.list?.let {
+            notificationRepository.insertNotifications(it)
+            NotificationPollBroadcastReceiver.onNotificationsComplete(context, it)
+        }
+    }
+
+    private fun assertLoggedIn() {
+        // Attempt to get a dummy CSRF token, which should automatically re-log us in explicitly,
+        // and should automatically log us out if the credentials are no longer valid.
+        CsrfTokenClient(WikipediaApp.getInstance().wikiSite).token
+            .subscribeOn(Schedulers.io())
+            .subscribe()
+    }
+
+    companion object {
+        var pollNotificationService = PollNotificationService()
+
+        fun schedulePollNotificationJob(context: Context) {
+            val serviceComponent = ComponentName(context, pollNotificationService.javaClass)
+            val builder = JobInfo.Builder(0, serviceComponent)
+            val jobScheduler = context.getSystemService(Context.JOB_SCHEDULER_SERVICE) as JobScheduler
+            jobScheduler.schedule(builder.build())
+        }
+         fun pollNotificationJobFinished() {
+            pollNotificationService.jobFinished(pollNotificationService.jobParameters, false)
```

Comment:
You're finishing the job of a Service class that was never created by the OS, so this will not do anything.